be/src/storage/rowset/segment_iterator.cpp

```diff
@@ -1277,6 +1488,37 @@ Status SegmentIterator::_do_get_next(Chunk* result, vector<rowid_t>* rowid) {
         chunk = _context->_adapt_global_dict_chunk.get();
     }
 
+    if (_use_vector_index && !_use_ivfpq) {
+        std::shared_ptr<FloatColumn> distance_column = FloatColumn::create();
+        vector<rowid_t> rowids;
+        for (const auto& id : _first_rowids) {
+            auto it = _id2distance_map.find(id);
+            if (it != _id2distance_map.end()) {
+                rowids.emplace_back(it->first);
+            }
+        }
+        if (rowid != nullptr) {
```

Comment:
`rowid` is an argument of `do_get_next`. And for query, `rowid` is always `nullptr`.
Only for compaction, `rowid` is non-`nullptr`.